iphone/Maps/UI/Settings/MWMTTSSettingsViewController.mm

```diff
@@ -52,6 +56,7 @@ virtual void SelectCell(UITableView * /* tableView */, NSIndexPath * /* indexPat
     [cell configWithDelegate:static_cast<id<SettingsTableViewSwitchCellDelegate>>(controller)
                        title:L(@"pref_tts_enable_title")
                         isOn:[MWMTextToSpeech isTTSEnabled]];
+    voiceInstructionsCell = cell;
```

Comment:
Using global vars in such a strange way doesn't look right. Is BuildCell called more than once?

Reply:
This code is trying to replicate the same structure of `MWMSettingsViewController.m` where a reference is stored to each cell to be able to distinguish between them in the `switchCell` callback